src/language_server/vfs.rs

```diff
@@ -0,0 +1,153 @@
+use std::collections::HashMap;
+use std::io;
+use std::io::ErrorKind;
+use std::sync::RwLock;
+
+use super::document::Document;
+
+use reqwest::blocking::Client;
```

Comment:
I did not realise this would block the event loop.  I was working on a threaded mental model so thought that blocking client would be fine.  Async client would be preferable in this case.